language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -311,7 +312,7 @@ class DottyLanguageServer extends LanguageServer
     val syms = Interactive.enclosingSourceSymbols(path, pos)
     val newName = params.getNewName
     val includes =
-      Include.references | Include.definitions | Include.linkedClass | Include.overriding
+      Include.references | Include.definitions | Include.linkedClass | Include.overriding | Include.imports
```

Comment:
What if I'm trying to rename  a tree whose name comes from a renamed import ?

Reply:
It'll behave the same as IntelliJ: The trees that use the new name won't be changed, but the ones that use the original symbol will. Renaming the renaming (renaming `baz` to something else in `import foo.{bar => baz}`) is not supported. There's a test for that here: https://github.com/lampepfl/dotty/pull/4199/files#diff-658e0b824b11a8c704adb2d5d7b05a38R91

But it would be better to display a warning too...

Reply:
But what prevents us from renaming the renaming technically ?

Reply:
Just adding ` | Include.renamingImports` here wouldn't work because then we'd rename imports into `import foo.{bar => bar}`. I'm not sure how much more complicated it'll get, I will look into it.